### DISCUSSION TITLE
Typst watch compiles files that are not watched

### DISCUSSION THREAD

**Discussion by mihock** [typst/typst#2622] [General]
### Description

If  file a.typ is watched and file b.typ in the same directory is changed, file b.typ is compiled to pdf (although this file is not watched). To reproduce:

```sh
echo "This is file a.typ" > a.typ
echo "This is file b.typ" > b.typ
typst watch a.typ
```

Now, if file b.typ is changed, a file b.pdf is produced.

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Dherse** [comment]
Are you by any chance using typst-lsp? Because I can assure you the source code of the cli doesn't allow it to produce more than one file.

**Reply by alerque** [reply]
The OP's example clearly doesn't involve Typst-LSP. Also it checks out for me too on Linux with nothing other than creating the two files via the shell and then running the CLI. When you first run `typst watch` it creates a.pdf and then waits. If you then `touch b.typ` then suddenly b.pdf shows up.

**Reply by Enivex** [reply]
See https://github.com/typst/typst/issues/2623

I was a bit too quick here...

**Reply by mihock** [reply]
@Enivex: Thank you very, perhaps you were not to quick. I actually made the change in VS Code with a running LSP. Without LSP, b.pdf will not be produced. So it seems to be a bug in the LSP?

**Reply by Dherse** [reply]
It's a setting in the lsp which you can disable

**Reply by alerque** [reply]
@mlhock @Dherse The LSP may be doing this *too*, but the original case still happens **with no lsp involved**. I'm not even running an editor and don't have the lsp installed and I still get the extra file created.

**Reply by mihock** [reply]
Merci! So this was my error, sorry.

**Reply by mihock** [reply]
@alerque: For me, the problem was solved with the Typst LSP setting "typst-lsp.exportPdf": "never".

**Reply by alerque** [reply]
@mlhock It sounds like what you reported is not actually what you tested. You *reported* a minimum working example using some simple shell commands and no editor/lsp in the picture. Now it might be true that you fixed your editor/lsp issue with a setting, it is also notable that your shell-only example still shows this **is** actually a problem in the CLI. I replicated in on the shell with no LSP or editor running and just touching the file, so there is still something wrong here.

**Reply by Dherse** [reply]
I just tried on WSL and I cannot reproduce the issue you have @alerque. For me running it, it only ever creates `a.pdf` or do you mean it updates `a.pdf` even though it shouldn't?

**Reply by alerque** [reply]
No, I mean it actually creates a `b.pdf` when `b.typ` is touched, even while only `typst watch a.typ` was invoked and nothing in the files links them to each other.

Typst v0.9.0 on Linux.

**Reply by alerque** [reply]
And ... my testing might be faulty for other reasons. I have a `touch b.pdf` in my history somewhere (my shell history is not linear because it's merged from many sources) and I can't replicate this now when I just tried to script the full MWE. It's possible I screwed up the test by touching the output instead of the input. I'll try this again when I get a chance.
